content/200-concepts/100-components/01-prisma-schema/06-relations/07-referential-actions.mdx

```diff
@@ -8,6 +8,401 @@ preview: true
+| `onDelete` | `SetNull`          | `Cascade`           |
+| `onUpdate` | `Cascade`          | `Cascade`           |
+
+If you need to actually use the cascade behavior configured in the database, you _can_ use [`$raw`](../../prisma-client/raw-database-access) SQL queries to [delete multiple referenced records](../../prisma-client/crud#deleting-all-data-with-raw-sql--truncate) (such as users blog posts) and the deletion will be successful. This is because the Prisma Client will **not** perform runtime checks on raw queries.
+
+## Upgrade paths
+
+There are a couple of paths you can take when upgrading which will give different results depending on the desired outcome.
+
+The following assumes you have upgraded to 2.26.0 and [enabled the preview feature flag](#enable-the-referentialactions-preview-feature).
+
+### Using Introspection
+
+If you [Introspect](../introspection) your database **before modifying your data model**, the [<=2.25.0 default values](#the-2250-default-referential-actions) will be reflected in your prisma schema. 
+
+Of note is the default value of `onDelete`, which <=2.25.0 is `Cascade`. 
+
+<Admonition type="info">
+
+The [Prisma VSCode extension](https://marketplace.visualstudio.com/items?itemName=Prisma.prisma) (if installed) will automatically add `onDelete: Cascade` to any required relations defined in your prisma schema.
+
+</Admonition>
+
+#### A blog schema example
+
+```prisma highlight=4;add
+model Post {
+  id        Int     @id @default(autoincrement())
+  title     String
+  author    User    @relation(fields: [authorId], references: [id], onDelete: Cascade)
+  authorId  Int
+}
+
+model User {
+  id    Int    @id @default(autoincrement())
+  posts Post[]
+}
+```
+
+<Admonition type="alert">
+
+If you are using either the [`delete()`](../../../../guides/database/advanced-database-tasks/cascading-deletes/postgresql#91-validating-restrict) or [`deleteAll()`](../../prisma-client/crud#delete-all-records) methods, **cascading deletes will now be performed** as the `referentialAction` preview feature **removed the safety net in Prisma Client that previously prevented cascading deletes at runtime**. Be sure to check your code and make any adjustments accordingly.
+
+</Admonition>
+
+### Using Migration
+
+When running a [Migration](../../prisma-migrate) (or the [`prisma db push`](../../prisma-migrate/db-push) command) the [new defaults](#referential-action-defaults) will be applied to your database.
+
+<Admonition type="info">
+
+Unlike when you run an Introspect for the first time, the new referential actions clause and property, will **not** automatically be addded to your prisma schema by the Prisma VSCode extension. 
+You will have to manually add them if you wish to use anything other than the new defaults.
+
+</Admonition>
+
+Using referential actions is opt in and you are not required to add them to your prisma schema. If you choose to omit them, the [new defaults](#referential-action-defaults) will be used.
```

Comment:
There are two ideas here that I could be confusing. Is the intent to explain enabling the feature is opt-in? Or that specifying custom referential actions is optional?

If it's the latter, I'd recommend using `optional` instead of opt-in. Maybe we could also add a note that this can be done case by case e.g. you can customize the referential action on one single relation and leave the rest on the defaults by not specifying anything.

Reply:
The intention was `specifying custom referential actions is optional` but obviously wasn't clear 😅Will make the changes!